src/database/DataTypes/StocktakeItem.js

```diff
@@ -95,45 +95,54 @@ export class StocktakeItem extends Realm.Object {
    * @param  {number} quantity   Change in StocktakeItem counted quantity
    */
   setCountedTotalQuantity(database, quantity) {
-    let difference = quantity - this.countedTotalQuantity;
-    if (difference === 0) return;
-
     database.write(() => {
-      // Create a StocktakeBatch and ItemBatch if none exist
+      // If there are no batches, create one
       if (this.batches.length === 0) {
         this.createNewBatch(database);
       }
-      const isIncreasingQuantity = difference > 0;
-      const sortedBatches = this.batches.sorted('expiryDate', isIncreasingQuantity);
-
-      sortedBatches.some(stocktakeBatch => {
-        const batchTotalQuantity = stocktakeBatch.countedTotalQuantity;
 
-        let thisBatchChangeQuantity = 0;
-        if (isIncreasingQuantity) {
-          thisBatchChangeQuantity = Math.min(stocktakeBatch.snapshotTotalQuantity -
-                                            batchTotalQuantity, difference);
-
-          if (thisBatchChangeQuantity <= 0) return false;
-        } else {
-          thisBatchChangeQuantity = Math.min(batchTotalQuantity, -difference);
-          thisBatchChangeQuantity = -thisBatchChangeQuantity;
+      let difference = quantity - this.countedTotalQuantity;
+      // In case number is entered in Actual Quantity field, but it's the same as
+      // snapshot quantity, we want to remove 'Not Counted' placeholder
+      if (difference === 0) {
+        this.batches.forEach(stocktakeBatch => {
+          stocktakeBatch.countedTotalQuantity = stocktakeBatch.snapshotTotalQuantity;
```

Comment:
Would have thought it'd be cleaner as 
`stocktakeBatch.countedTotalQuantity = quantity;`
But it should literally be the same result and same performance as far as I know, so not really fussed. Your way kind of plays into what the getter for `difference` looks like on stocktakeBatch it could be argued.

Reply:
I think it's also meant to set countedTotalQuantity `for each stocktakeBatch` to snapshotTotalQuantity. i.e. if there are two ItemBatches for an item, batchA.totalQuantity = 5, batchB.totalQuantity = 10, when a stocktake is created, two stocktakeBatch will be made for that item with countedTotalQuantity = null and StocktakeBatchA.snapshotTotalQuantity = batchA.totalQuantity and StocktakeBatchB.snapshotTotalQuantity = batchB.totalQuantity. So if the quantity does not change but a number in the actual quantity field, we need to set countedTotalQuantity = snapshotTotalQuantity for each batch (which maybe different from actual quantity).

Reply:
Yup looking again that makes sense.

Reply:
Although does this break stocktaking at the batch level at all? A quick test would show.

Reply:
I've tested, seems all good

Reply:
Ages to pick this up, but this is the cause of #626

Emphasises the need for:
a. Better branch management for features
b. Comprehensive testing before releasing to users
c. Automated testing